src/controllers/user.controller.ts

```diff
@@ -760,6 +760,190 @@ export const removeNotificationToken = async (req: Request, res: Response) => {
   }
 };
 
+/**
+ * Adds an emergency contact to a given user's list of contacts.
+ *
+ * @param {Request} req - The Express request object containing the user ID and contact information.
+ * @param {Response} res - The Express response object to modify with success or error messages.
+ * @returns {Promise} A Promise that resolves when the contact has been added or rejects with an error.
+ */
+export const addEmergencyContact = async (req: Request, res: Response) => {
+  const userId = req.params?.userId;
+  const emergencyContact = req.body;
+
+  // Check if the provided userId is valid
+  if (!mongoose.Types.ObjectId.isValid(userId)) {
+    return res.status(400).send({ message: 'Invalid user ID!' });
+  }
+
+  // Check if the emergency contact is valid
+  const validationError = verifyKeys(
+    emergencyContact,
+    KeyValidationType.EMERGENCY_CONTACTS
+  );
+  if (validationError !== '') {
+    return res.status(400).send({ message: validationError });
+  }
+
+  const generatedId = new mongoose.Types.ObjectId();
+
+  try {
+    // Find the user in the database and add the emergency contact to their account
+    const targetUser = await User.findByIdAndUpdate(
+      { _id: userId },
+      {
+        $push: { emergencyContacts: { ...emergencyContact, _id: generatedId } },
+      },
+      { new: true }
+    );
+
+    // Check if the user exists
+    if (targetUser === null) {
+      return res.status(400).send({
+        message: 'User does not exist!',
+      });
+    }
```

Comment:
I don't think this will ever happen because of the `ObjectId` validation line 775